sktime/forecasting/arch.py

```diff
@@ -0,0 +1,392 @@
+    ...    vol="Garch",
+    ...    p=1,
+    ...    q=1,
+    ...    dist="Normal",
+    ...)
+    >>> forecaster.fit(y)
+    ARCH(...)
+    >>> y_pred = forecaster.predict(fh=[1,2,3])
+    """
+
+    _tags = {
+        "scitype:y": "univariate",
+        "y_inner_mtype": ["np.ndarray", "pd.Series"],
+        "X_inner_mtype": ["np.ndarray", "pd.DataFrame"],
+        "requires-fh-in-fit": False,
+        "capability:pred_int": False,
+    }
+
+    _fitted_param_names = (
+        "update_freq",
+        "disp",
+        "starting_values",
+        "cov_type",
+        "show_warning",
+        "first_obs",
+        "last_obs",
+        "tol",
+        "options",
+        "backcast",
+    )
+
+    def __init__(
+        self,
+        mean="Constant",
+        lags=0,
+        vol="Garch",
+        p=1,
+        o=0,
+        q=1,
+        power=2.0,
+        dist="Normal",
+        hold_back=None,
+        rescale=None,
+        update_freq=0,
+        disp="off",
+        starting_values=None,
+        cov_type=None,
+        show_warning=None,
+        first_obs=None,
+        last_obs=None,
+        tol=None,
+        options=None,
+        backcast=None,
+        params=None,
+        start=None,
+        align=None,
+        method="analytic",
+        simulations=None,
+        rng=None,
+        random_state=None,
+        reindex=None,
+    ):
+
+        _check_soft_dependencies("arch", severity="error", object=self)
+
+        self.mean = mean
+        self.lags = lags
+        self.vol = vol
+        self.p = p
+        self.o = o
+        self.q = q
+        self.power = power
+        self.dist = dist
+        self.hold_back = hold_back
+        self.rescale = rescale
+        self.update_freq = update_freq
+        self.disp = disp
+        self.starting_values = starting_values
+        self.cov_type = cov_type
+        self.show_warning = show_warning
+        self.first_obs = first_obs
+        self.last_obs = last_obs
+        self.tol = tol
+        self.options = options
+        self.backcast = backcast
+        self.params = params
+        self.start = start
+        self.align = align
+        self.method = method
+        self.simulations = simulations
+        self.rng = rng
+        self.random_state = random_state
+        self.reindex = reindex
+        self._forecaster = None
+
+        super(ARCH, self).__init__()
+
+    def _fit(self, y, X=None, fh=None):
+        """Fit the training data to the estimator.
+
+        Parameters
+        ----------
+        y : guaranteed to be of a type in np.ndarray or pd.Series
+            Time series to which to fit the forecaster, guaranteed to have a
+            single column/variable
+        X : optional (default=None)
+            guaranteed to be of a type in np.ndarray or pd.DataFrame
+            Exogeneous time series to fit to.
+
+        Returns
+        -------
+        self : returns an instance of self
+        """
+        from arch import arch_model as _ARCH
```

Comment:
Should this import occur inside the _fit function?

Reply:
no, because it is a soft dependency.

All soft dependency imports should be isolated in a way like this, so importing the module does not error out under the minimal dependency set (which excludes soft dependencies).

All estimators should also be aware of the soft dependencies in them, via the `python_dependencies` tag.